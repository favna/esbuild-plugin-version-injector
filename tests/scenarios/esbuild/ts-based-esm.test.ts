import esbuild, { type BuildOptions } from 'esbuild';
import type { PluginOptions } from '../../../dist';
import { assertFileContent, buildAbsolutePath, createEsbuildConfig } from '../../common/util';

function createConfig(pluginOptions?: PluginOptions): BuildOptions {
  return createEsbuildConfig(
    {
      format: 'esm',
      outExtension: {
        '.js': '.mjs'
      },
      entryPoints: [buildAbsolutePath('../fixtures/build-in/typescript.mts')]
    },
    pluginOptions
  );
}

describe('TS based ESM', () => {
  test('GIVEN no plugin options THEN injects version', async () => {
    const config = createConfig();

    await esbuild.build(config);

    await assertFileContent('../fixtures/build-out/esbuild/typescript.mjs');
  });

  test('GIVEN versionOrCurrentDate = "current-date" THEN injects current date', async () => {
    const config = createConfig({
      versionOrCurrentDate: 'current-date'
    });

    await esbuild.build(config);

    await assertFileContent('../fixtures/build-out/esbuild/typescript.mjs');
  });
});
