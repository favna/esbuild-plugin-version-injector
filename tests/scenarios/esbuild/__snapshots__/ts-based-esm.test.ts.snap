// Vitest Snapshot v1

exports[`TS based ESM > GIVEN no plugin options THEN injects version 1`] = `
"// tests/fixtures/build-in/typescript.mts
var version = \\"1.0.2\\";
export {
  version
};
"
`;

exports[`TS based ESM > GIVEN versionOrCurrentDate = "current-date" THEN injects current date 1`] = `
"// tests/fixtures/build-in/typescript.mts
var version = \\"2022-02-01T14:30:30.000Z\\";
export {
  version
};
"
`;
